lib/spartacus.ml

```diff
@@ -0,0 +1,118 @@
+open Rresult
+open Ptt_tuyau.Lwt_backend
+open Lwt.Infix
+
+module Make
+    (Random : Mirage_random.S)
+    (Time : Mirage_time.S)
+    (Mclock : Mirage_clock.MCLOCK)
+    (Pclock : Mirage_clock.PCLOCK)
+    (Resolver : Ptt.Sigs.RESOLVER with type +'a io = 'a Lwt.t)
+    (Stack : Tcpip.Stack.V4V6) =
+struct
+  include Ptt_tuyau.Make (Stack)
+
+  let src = Logs.Src.create "nec"
+
+  module Log : Logs.LOG = (val Logs.src_log src)
+
+  module Random = struct
+    type g = Random.g
+    type +'a io = 'a Lwt.t
+
+    let generate ?g buf =
+      let len = Bytes.length buf in
+      let raw = Random.generate ?g len in
+      Cstruct.blit_to_bytes raw 0 buf 0 len
+      ; Lwt.return ()
+  end
+
+  module Signer =
```

Comment:
Instead of `Signer`, it will be nice to rename it `Filter` :+1: to be not confuse about the goal of this module.